### PR TITLE
Multimodal autoencoder

### PR DESCRIPTION
**Pull request by OlivierBernard**

### CODE REVIEW

File: collections/_posts/2023-07-17-Mutlimodal-AutoEncoder.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+##### Vision encoder
+
+$$X^{\nu} \in \mathbb{R}^{(N+1) \times D} \,=\, \left[ p_I; p_1 E^{\nu}; \cdots; p_N E^{\nu} \right]\,+\,E^{\nu}_{pos}$$
+
+- Each image $$I \in \mathbb{R}^{H \times W \times C}$$ is divided into $$N$$ patches $$\{ p_1,\cdots,p_N \}$$
+
+- $$E^{\nu} \in \mathbb{R}^{P^2 \times D}$$ is the projection matrix into the patch embeddings
+
+- $$p_I \in \mathbb{R}^{D}$$ used for the aggregation of visual information
```
Comment (nathanpainchaud) [inline collections/_posts/2023-07-17-Mutlimodal-AutoEncoder.md:92]: Missing word:
```suggestion
- $$p_I \in \mathbb{R}^{D}$$ is used for the aggregation of visual information
```

File: collections/_posts/2023-07-17-Mutlimodal-AutoEncoder.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+- $$X^{\nu}$$ is fed into a transformer model with $$N_{\nu}$$ transformer blocks to obtain the contextualized image representation $$H^{\nu} \in \mathbb{R}^{(N+1) \times D} \,=\, \left[ h^{\nu}_I; h^{\nu}_1; \cdots; h^{\nu}_N \right]$$
+
+&nbsp;
+
+##### Language encoder
+
+$$X^{l} \in \mathbb{R}^{(M+2) \times D} \,=\, \left[ w_T; w_1 E^{l}; \cdots; w_M E^{l}; w_{SEP} \right]\,+\,E^{l}_{pos}$$
+
+- Each input text is tokenize to subword tokens $${w_1,\cdots;w_M}$$ by WordPiece, where tokens $$w_m \in \mathbb{R}^{V}$$ are represented in one-hot form and $$V$$ is the vocabulary size
```
Comment (nathanpainchaud) [inline collections/_posts/2023-07-17-Mutlimodal-AutoEncoder.md:102]: Conjugation:
```suggestion
- Each input text is tokenized to subword tokens $${w_1,\cdots;w_M}$$ by WordPiece, where tokens $$w_m \in \mathbb{R}^{V}$$ are represented in one-hot form and $$V$$ is the vocabulary size
```

File: collections/_posts/2023-07-17-Mutlimodal-AutoEncoder.md [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+##### Masking scheme
+
+- the authors used random sampling with a much greater masking ratio for images (i.e. $$75\%$$) than for texts (i.e. $$15\%$$). This is justified by the fact that images are redundant while languages are information-dense
+
+&nbsp;
+
+##### Representation selection for reconstruction
+
+- Images and texts are abstracted at different levels, with pixels having a lower semantic level than token of texts.
```
Comment (nathanpainchaud) [inline collections/_posts/2023-07-17-Mutlimodal-AutoEncoder.md:120]: Phrasing:
```suggestion
- Images and texts are abstracted at different levels, with pixels having a lower semantic level than text tokens.
```

File: collections/_posts/2023-07-17-Mutlimodal-AutoEncoder.md [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+
+![](/collections/images/multimodal_autoencoder/qualitative-results.jpg)
+
+&nbsp;
+
+# Conclusions
+
+* Image/text coupling for medical data analysis looks like a promising way forward
+
+* Pre-training in a self-supervised way using maskink strategy appears to be relevant
```
Comment (nathanpainchaud) [inline collections/_posts/2023-07-17-Mutlimodal-AutoEncoder.md:230]: Typo:
```suggestion
* Pre-training in a self-supervised way using a masking strategy appears to be relevant
```

File: collections/_posts/2023-07-17-Mutlimodal-AutoEncoder.md [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+
+* Exploiting embedings at different levels of abstraction for images and text would seem to be a good approach
```
Comment (nathanpainchaud) [inline collections/_posts/2023-07-17-Mutlimodal-AutoEncoder.md:232]: Typo:
```suggestion
* Exploiting embeddings at different levels of abstraction for images and text would seem to be a good approach
```

**Review by nathanpainchaud — CHANGES_REQUESTED** [review]
Great post as always! Just a few typos and phrasing to fix.

File: collections/_posts/2023-07-17-Mutlimodal-AutoEncoder.md
Comment (nathanpainchaud) [inline collections/_posts/2023-07-17-Mutlimodal-AutoEncoder.md:1]: There's a typo in the name of the fle as well :sweat_smile: _Mutlimodal_ -> _Multimodal_

**Review by nathanpainchaud — CHANGES_REQUESTED** [review]
I missed this typo when simply reading the post.

File: collections/_posts/2023-07-17-Mutlimodal-AutoEncoder.md
Comment (OlivierBernard) [reply collections/_posts/2023-07-17-Mutlimodal-AutoEncoder.md:1]: I change the filename, thanks !

**Review by nathanpainchaud — APPROVED** [review]

**Pull request merged**